packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -98,6 +96,7 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
 
       return new Bluebird((resolve, reject) => {
         const cleanup = () => {
+          Cypress.backend('cross:origin:finished', location.originPolicy)
```

Comment:
I don't think we do this right now, but do we need to cleanup when `uncaught:error`s happen here as well to prevent any type of dangling remote state?

Reply:
In order to prevent dangling remote state I'm resetting the remote state before each test and in a few other scenarios as well.